flink/src/test/java/org/apache/iceberg/flink/TestFlinkTableSource.java

```diff
@@ -90,17 +103,418 @@ public void testLimitPushDown() {
+
+    List<Object[]> expectedBetween = Lists.newArrayList();
+    expectedBetween.add(new Object[] {1, "a"});
+    expectedBetween.add(new Object[] {2, "b"});
+    Assert.assertArrayEquals("Should produce the expected record", resultBetween.toArray(), expectedBetween.toArray());
+
+    Assert.assertEquals("Should create only one scan", 1, scanEventCount);
+    Assert.assertEquals("should contains the push down filter", "(ref(name=\"id\") >= 1 and ref(name=\"id\") <= 2)",
+        lastScanEvent.filter().toString());
+  }
+
+  @Test
+  public void testFilterPushDownNotBetween() {
+    String sqlNotBetween = String.format("SELECT * FROM %s WHERE id  NOT BETWEEN 2 AND 3 ", TABLE_NAME);
+    String explainNotBetween = getTableEnv().explainSql(sqlNotBetween);
+    String expectedFilter = "(ref(name=\"id\") < 2 or ref(name=\"id\") > 3)";
+    Assert.assertTrue("explain should contains the push down filter", explainNotBetween.contains(expectedFilter));
+
+    List<Object[]> resultNotBetween = sql(sqlNotBetween);
+    Assert.assertEquals("should have 1 record", 1, resultNotBetween.size());
+    Assert.assertArrayEquals("the not between should produce the expected record", resultNotBetween.get(0),
+        new Object[] {1, "a"});
+
+    Assert.assertEquals("Should create only one scan", 1, scanEventCount);
+    Assert.assertEquals("should contains the push down filter", lastScanEvent.filter().toString(), expectedFilter);
+  }
+
+  @Test
+  public void testFilterPushDownLike() {
+    String sqlLike = "SELECT * FROM " + TABLE_NAME + " WHERE data LIKE 'a%' ";
+    String explainLike = getTableEnv().explainSql(sqlLike);
+    String expectedFilter = "ref(name=\"data\") startsWith \"\"a\"\"";
+    Assert
+        .assertTrue("the like sql explain should contains the push down filter", explainLike.contains(expectedFilter));
+
+    List<Object[]> resultLike = sql(sqlLike);
+    Assert.assertEquals("should have 1 record", 1, resultLike.size());
+    Assert.assertArrayEquals("the like result should produce the expected record", resultLike.get(0),
+        new Object[] {1, "a"});
+    Assert.assertEquals("Should create only one scan", 1, scanEventCount);
+    Assert.assertEquals("should contains the push down filter", lastScanEvent.filter().toString(), expectedFilter);
+
+    // not push down
+    String sqlNoPushDown = "SELECT * FROM " + TABLE_NAME + " WHERE data LIKE '%a%' ";
+    String explainNoPushDown = getTableEnv().explainSql(sqlNoPushDown);
+    Assert.assertFalse("explain should not contains FilterPushDown",
+        explainNoPushDown.contains(expectedFilterPushDownExplain));
+  }
+
+  @Test
+  public void testFilterPushDown2Literal() {
+    String sql2Literal = String.format("SELECT * FROM %s WHERE 1 > 0 ", TABLE_NAME);
+    String explain2Literal = getTableEnv().explainSql(sql2Literal);
+    Assert.assertFalse("explain should not contains FilterPushDown",
+        explain2Literal.contains(expectedFilterPushDownExplain));
+  }
```

Comment:
@zhangjun0x01, I think this suite also needs to test the behavior of a few more cases to ensure that the assumptions about Flink's parser hold. Even if these result in parser errors, those are good tests to have to validate that if Flink ever does support parsing the expressions, Iceberg will do the right thing.

* `double_col = NaN`
* `double_col <> NaN`
* All inequalities with NaN, like `double_col < NaN`
* All inequalities with null, like `data < null`

Reply:
We can add the test case `All inequalities with null, like data < null`.
 in spark and hive,we can use sql `select float('NaN')` to get a `NaN` ,but the `NaN` and `Infinity` are not supported by flink ([doc](https://ci.apache.org/projects/flink/flink-docs-stable/dev/table/types.html)), we can add a `TODO` in the class to do the test in the futher

Reply:
Yes, if those are not supported, then we should have tests that assert a parse error. We want the tests to start failing if we update to a version that does support `NaN` in the parser and it is used in an expression.

Reply:
yes,I will add the test case to assert parse error, when we upgrade to a version of flink that supports `NaN`, we will not forget to modify these test cases that fail.

Reply:
I add the test case ([link](https://github.com/zhangjun0x01/iceberg/blob/filterPushDown/flink/src/test/java/org/apache/iceberg/flink/TestFlinkFilters.java#L300)) ,and add the sql test case ([link](https://github.com/zhangjun0x01/iceberg/blob/filterPushDown/flink/src/test/java/org/apache/iceberg/flink/TestFlinkTableSource.java#L580))